src/event/metric.rs

```diff
@@ -360,6 +379,112 @@ impl Display for Metric {
     }
 }
 
+impl Object for Metric {
+    fn insert(&mut self, path: &remap::Path, value: remap::Value) -> Result<(), String> {
+        if path.is_root() {
+            return Err("cannot set root path for a metric event".into());
+        }
+
+        match path.segments() {
+            [Segment::Field(tags), Segment::Field(field)] if tags.as_str() == "tags" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                self.set_tag_value(
+                    field.as_str().to_owned(),
+                    String::from_utf8_lossy(value).into_owned(),
+                );
+                Ok(())
+            }
+            [Segment::Field(name)] if name.as_str() == "name" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                self.name = String::from_utf8_lossy(value).into_owned();
+                Ok(())
+            }
+            [Segment::Field(namespace)] if namespace.as_str() == "namespace" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                self.namespace = Some(String::from_utf8_lossy(value).into_owned());
+                Ok(())
+            }
+            [Segment::Field(timestamp)] if timestamp.as_str() == "timestamp" => {
+                let value = value.as_timestamp().ok_or("value must be a timestamp")?;
+                self.timestamp = Some(*value);
+                Ok(())
+            }
+            [Segment::Field(kind)] if kind.as_str() == "kind" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                match String::from_utf8_lossy(value).as_ref() {
+                    "absolute" => {
+                        self.kind = MetricKind::Absolute;
+                        Ok(())
+                    }
+                    "incremental" => {
+                        self.kind = MetricKind::Incremental;
+                        Ok(())
+                    }
+                    _ => Err("metric kind must be `absolute` or `incremental`".into()),
+                }
+            }
+            _ => Err("invalid path".into()),
```

Comment:
We probably want a bit more of a descriptive error message here.

One way to tackle this is to have a static list of whitelisted paths, and have this error report a message such as "got invalid path X, expected one of Y, Z"